corehq/apps/app_manager/models.py

```diff
@@ -1527,6 +1531,11 @@ def _get_media_by_language(self, media_attr, lang, strict=False):
         """
         assert media_attr in ('media_image', 'media_audio')
 
+        if self.media_image_languages_linked and media_attr == 'media_image':
+            lang = self.get_app().default_language
+        if self.media_audio_languages_linked and media_attr == 'media_audio':
+            lang = self.get_app().default_language
+
```

Comment:
the line above already asserts that `media_attr` is in `('media_image', 'media_audio')`. So i think this can be rewritten as
```
if self. media_image_languages_linked or self.media_audio_languages_linked:
    lang = self.get_app().default_language
```
only.

Reply:
this method gets called once for `media_image` and once for `media_audio`, and returns the media-path for that particular media item. If I use the `if` statement you are suggesting, if the `media_audio_languages_linked` item is selected, it will return the default language for images too, which is not the right behaviour. I considered refactoring this area of code as there is a lot of indirection and confusing things like this, but decided it was a bit out of scope considering most of this stuff hasn't been touched since like 2013, and is unlikely to be touched again soon.

Reply:
thats tight. thanks.